src/core/hash.zig

```diff
@@ -1,69 +1,48 @@
 const std = @import("std");
 const Sha256 = std.crypto.hash.sha2.Sha256;
-const Allocator = std.mem.Allocator;
 const base58 = @import("base58-zig");
-const CompareOperator = std.math.CompareOperator;
 
 pub const HASH_SIZE: usize = 32;
 
 pub const Hash = struct {
     data: [HASH_SIZE]u8,
 
-    const Self = @This();
-
-    pub fn default() Self {
-        return .{ .data = [_]u8{0} ** HASH_SIZE };
+    pub fn default() Hash {
+        return .{ .data = .{0} ** HASH_SIZE };
     }
-    // used in tests
-    pub fn random() Self {
-        var seed = @as(u64, @intCast(std.time.milliTimestamp()));
-        var rand = std.rand.DefaultPrng.init(seed);
-        var data: [HASH_SIZE]u8 = undefined;
 
-        for (0..HASH_SIZE) |i| {
-            data[i] = rand.random().int(u8);
-        }
-        return Self{
-            .data = data,
-        };
+    pub fn generateSha256Hash(bytes: []const u8) Hash {
+        var data: [HASH_SIZE]u8 = undefined;
+        Sha256.hash(bytes, &data, .{});
+        return .{ .data = data };
     }
 
-    pub fn generateSha256Hash(bytes: []const u8) Self {
-        var hash = Hash{
-            .data = undefined,
-        };
-        Sha256.hash(bytes, &hash.data, .{});
-        return hash;
+    pub fn extendAndHash(id: Hash, val: []const u8) Hash {
+        var hasher = Sha256.init(.{});
+        hasher.update(&id.data);
+        hasher.update(val);
+        return .{ .data = hasher.finalResult() };
     }
 
-    pub fn cmp(a: *const Self, b: *const Self) CompareOperator {
-        for (0..HASH_SIZE) |i| {
-            if (a.data[i] > b.data[i]) {
-                return .gt;
-            } else if (a.data[i] < b.data[i]) {
-                return .lt;
-            }
+    pub fn order(a: *const Hash, b: *const Hash) std.math.Order {
+        for (a.data, b.data, 0..HASH_SIZE) |a_byte, b_byte, _| {
```

Comment:
can you remove this unused `0..HASH_SIZE`

Reply:
sure thing, come up right now